base/src/main/java/io/spine/base/Environment.java

```diff
@@ -104,6 +107,36 @@
  * }
  * </pre>
  *
+ * <h1>Caching</h1>
+ * <p>{@code Environment} caches the {@code EnvironmentType} once its calculated.
```

Comment:
For the sake of readability of this `.java` file, please add an empty line above.